### PR TITLE
feat: Verifier in parallel

### PR DESCRIPTION
**Pull request by JonathanStarup**
Relates to  #7440

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 1]
```diff
@@ -28,7 +28,7 @@ object Verifier {
 
   def run(root: Root)(implicit flix: Flix): Root = flix.phase("Verifier") {
     if (!flix.options.xnoverify) {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:30]: Could we change this to early return?

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 2]
```diff
@@ -28,7 +28,7 @@ object Verifier {
...
-      root.defs.values.foreach(d => visitDef(d)(root))
+      ParOps.parMap(root.defs.values)(visitDef(_)(root))
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:31]: Is there not a better parMap thing or  do we have to construct the .values thing?

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:31]: Its not super clear from the docs, but values is `Iterable` and is based on `valuesIterator` so I'm pretty sure its not materialized.

There is no wellsuited ParOps things - it doesn't have foreach style methods

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala (hunk 1 above)
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:30]: I wrote it like this to be consistent with Optimizer - I can change both? They can also both just use if-else

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:30]: Yes, please

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:31]: Ok

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Woops. Build issue.

**Pull request merged**